wikimedia/wikipedia-ios
# Fix iPadOS18 push navigation bar bugs

**Phabricator:** https://phabricator.wikimedia.org/T375445

### Notes
This fixes the strange navigation bar layout we were seeing when initially pushing to Watchlist and Image Recommendations on iPadOS18.

The reason for this is that most other views in the app use a custom navigation bar, but Watchlist and Image Recommendations do NOT use a custom navigation bar. Pushing from a custom navigation bar context (i.e. Explore) causes the system navigation bar to go haywire initially (but only for iPadOS18, so this is likely related to the floating tab bar). Asking the view controller's navigation controller to relayout after appearance fixes it.

### Test Steps
1. On iPadOS18, install app and log in. From the new profile view, tap Watchlist. Confirm navigation bar looks okay after pushing on Watchlist (note you will see a layout "jump", which isn't great, but still an improvement to what it was before).
2. Also tap the Suggested Edits Image Recommendations card in the Explore feed. Confirm navigation bar looks okay after pushing on Image Recommendations.
